presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/StaticHiveCluster.java

```diff
@@ -67,12 +82,27 @@ public StaticHiveCluster(List<URI> metastoreUris, String metastoreUsername, Hive
     public HiveMetastoreClient createMetastoreClient()
     {
         List<HostAndPort> metastores = new ArrayList<>(addresses);
-        Collections.shuffle(metastores.subList(1, metastores.size()));
+        metastores.sort(Comparator.comparingLong(m -> concurrentMap.get(m).getBackoffDelayNanos()));
```

Comment:
`concurrentMap.get(m).getBackoffDelayNanos()` is not a function, as it can return different values on subsequent calls. I think this renders `metastores.sort` result indeterministic. Instead, you should pull the value for each address once, then sort.. You can do this in Steamy fashion.